### PR TITLE
Add requirements updates from pyup

### PR DESCRIPTION
**Pull request by Emantor**
I squashed the pyup pull request and added my own fixes. Afterwards we can enable pyup for labgrid.

### CODE REVIEW

**Comment by Emantor** [comment]
Ah, how do we solve the DCO checks? Is there a workaround for a bot feature like this?

**Comment by Emantor** [comment]
According to probot/dco#21 there is an issue for DCO and bot support, but nobody started work on it yet.

File: dev-requirements.txt [hunk 1]
```diff
@@ -1,22 +1,15 @@
...
-pyudev
-yapf
-requests
-autobahn
-xmodem>=0.4.5
+PyYAML==3.12
+pyModbusTCP==0.1.5
+pysnmp==4.4.4
+pysnmp-mibs==0.1.6
+pytest==3.4.0
```
Comment (jluebbe) [inline dev-requirements.txt]: pytest is already part of requirements.txt.

File: dev-requirements.txt [hunk 2]
```diff
@@ -1,22 +1,15 @@
...
-pytest-mock
-pytest-pylint
-pytest-runner
-onewire>=0.2
-pyudev
-yapf
-requests
-autobahn
-xmodem>=0.4.5
+PyYAML==3.12
```
Comment (jluebbe) [inline dev-requirements.txt]: PyYAML is needed by default.

**Comment by Emantor** [comment]
Updated.

**Comment by Emantor** [comment]
I split out the optional features and documented them in the installation section as well.

**Pull request merged**